backend/battling/api/serializers.py

```diff
@@ -66,3 +79,80 @@ def create(self, validated_data):
 
         set_up_battle_teams_and_send_invite_email(instance)
         return instance
+
+
+class BattleTeamSerializer(serializers.ModelSerializer):
+    team = TeamSerializer(read_only=True)
+    pokemons = PokemonSerializer(many=True, read_only=True)
+    pokemons_ids = serializers.PrimaryKeyRelatedField(
+        source="pokemons",
+        queryset=Pokemon.objects.all(),
+        many=True,
+    )
+
+    class Meta:
+        model = PokemonTeam
+        fields = (
+            "id",
+            "team",
+            "pokemons",
+            "pokemons_ids",
+        )
+
+    def validate(self, attrs):
+        pokemon_list = attrs["pokemons"]
+        if len(pokemon_list) != 3:
+            raise serializers.ValidationError("ERROR: All fields are required.")
+
+        team_has_repeated_pokemon = has_repeated_pokemon(pokemon_list)
+        if team_has_repeated_pokemon:
+            raise serializers.ValidationError("You can't choose the same Pokemon more than once.")
+
+        is_pokemon_sum_valid = is_team_valid(pokemon_list)
+
+        if not is_pokemon_sum_valid:
+            raise serializers.ValidationError("ERROR: Your pokemons sum more than 600 points.")
+
+        return attrs
+
+    def _has_both_teams(self, battle):
+        creator = (
+            Team.objects.filter(battle=battle, trainer=battle.creator)
+            .prefetch_related("pokemons")
+            .get()
+        )
+        creator_pokemons = creator.pokemons.all()
+
+        opponent = (
+            Team.objects.filter(battle=battle, trainer=battle.opponent)
+            .prefetch_related("pokemons")
+            .get()
+        )
+        opponent_pokemons = opponent.pokemons.all()
+
+        return creator_pokemons and opponent_pokemons
```

Comment:
This method should be in a separated place, since it can be used on the serializer and on the form